Dockerfile-production

```diff
@@ -0,0 +1,63 @@
+FROM ruby:2.6.4-alpine AS builder
+
+# Build base for gem's native extensions
+# tzdata for ruby timezone data
+# gcompat for ffi to load LSC
+RUN echo "http://dl-4.alpinelinux.org/alpine/edge/community" >> /etc/apk/repositories \
+    && apk update \
+    && apk add -u --no-cache build-base tzdata gcompat git postgresql-dev bash yarn \
+    && rm -rf /var/cache/apk/*
+
+ENV LANG C.UTF-8
+ENV GEM_HOME /bundle
+ENV BUNDLE_PATH $GEM_HOME
+ENV BUNDLE_APP_CONFIG $BUNDLE_PATH
+ENV BUNDLE_BIN $BUNDLE_PATH/bin
+# Add bundle dir to path to be able to access commands outside of `bundle exec`
+ENV PATH /app/bin:$BUNDLE_BIN:$PATH
+
+ENV RAILS_ENV=production
+ENV NODE_ENV=production
+
+RUN gem update --system
+
+RUN mkdir -p /app
+WORKDIR /app
+
+COPY Gemfile* package.json yarn.lock ./
+
+RUN bundle install --without development:test \
+    && rm -rf $BUNDLE_PATH/cache/*.gem \
+    && find $BUNDLE_PATH/gems/ -name "*.c" -delete \
+    && find $BUNDLE_PATH/gems/ -name "*.o" -delete
+
+RUN yarn install
+COPY . .
+RUN DATABASE_URL=nulldb://host/db SECRET_KEY_BASE="$(bundle exec rails secret)" REDIS_URL=redis://redis:6379/1 \
+    bin/rails assets:precompile
+
+RUN rm -rf node_modules/ tmp/cache app/assets vendor/assets spec
+
+#############
+
+FROM ruby:2.6.4-alpine
```

Comment:
![tim-and-eric-mind-blown](https://user-images.githubusercontent.com/438911/65469254-a1b52f80-de1b-11e9-94ad-8197abfea5bb.gif)

Reply:
A quick breakdown is you use the first to pull in all dependencies and build them (a few more packages in the apk install).  Then once they're built you don't need the compiler, and in production we don't need anything from node_modules since its all been compiled so we can throw it all away and start in a clean fresh build.  Then we just copy over all the compiled assets and gems and we're good to go!